public/app/plugins/panel/xychart/scatter.ts

```diff
@@ -115,18 +115,20 @@ function getScatterSeries(
   if (dims.pointColorIndex) {
     const f = frames[frameIndex].fields[dims.pointColorIndex];
     if (f) {
-      const disp =
-        f.display ??
-        getDisplayProcessor({
-          field: f,
-          theme: config.theme2,
-        });
       pointColorMode = getFieldColorModeForField(y);
       if (pointColorMode.isByValue) {
         const index = dims.pointColorIndex;
         pointColor = (frame: DataFrame) => {
-          // Yes we can improve this later
-          return frame.fields[index].values.map((v) => disp(v).color!);
+          const field = frame.fields[index];
+
+          if (field.state?.range) {
+            // this forces local min/max recalc, rather than using global min/max from field.state
+            field.state.range = undefined;
+          }
+
+          field.display = getDisplayProcessor({ field, theme: config.theme2 });
+
+          return field.values.map((v) => field.display!(v).color!); // slow!
```

Comment:
this is the "easy" fix -- i assume we should do that here and backport to 10x, but try something better for 10.1?  This seems like a place where a fast lookup pattern is better!

Reply:
we will standardize on quantized/discrete palettes soon, and likely migrate a lot of this to the same interpolation strategy we have in Heatmap. so, yes.